Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs

```diff
@@ -318,18 +318,16 @@ public IWebElement LocateElement(Act act)
                 return mSeleniumDriver.LocateElement(act);
             }
 
-            eLocateBy locateBy = act.LocateBy;
+            eLocateBy locateBy = act is ActUIElement ? (act as ActUIElement).ElementLocateBy : act.LocateBy;
             IWebElement elem = null;
-
+            string locateValue = act is ActUIElement ? (act as ActUIElement).ElementLocateValue : act.LocateValue;
             switch (locateBy)
             {
                 case eLocateBy.ByResourceID:
-                    {
-                        elem = Driver.FindElementById(act.LocateValue);
+                        elem = Driver.FindElementById(locateValue);
                         break;
-                    }
                 default:
-                    elem = mSeleniumDriver.LocateElement(act);
+                    elem = Driver.FindElementByXPath(locateValue);
```

Comment:
why default is by Xpath and not by Selenium? it will impact execution flows, no? how locate ByName, ByID, etc will be performed for Web mobile flows?

Reply:
for Web mobile flows, it is already handled befor this switch-case @ line #318